oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/AuthCodeHandler.java

```diff
@@ -370,19 +375,27 @@ private AuthorizationCode generateAuthCode(
             URI redirectUri,
             ClientSession clientSession,
             String clientSessionId,
-            Session session)
+            Session session,
+            OrchSessionItem orchSession)
             throws ClientNotFoundException, ProcessAuthRequestException {
         if (!orchestrationAuthorizationService.isClientRedirectUriValid(clientID, redirectUri)) {
             throw new ProcessAuthRequestException(400, ErrorResponse.ERROR_1016);
         }
         CredentialTrustLevel lowestRequestedCredentialTrustLevel =
                 VectorOfTrust.getLowestCredentialTrustLevel(clientSession.getVtrList());
+        var currentCredentialStrength = session.getCurrentCredentialStrength();
         if (isNull(session.getCurrentCredentialStrength())
                 || lowestRequestedCredentialTrustLevel.compareTo(
                                 session.getCurrentCredentialStrength())
                         > 0) {
             session.setCurrentCredentialStrength(lowestRequestedCredentialTrustLevel);
         }
+        if (isNull(currentCredentialStrength)
+                || lowestRequestedCredentialTrustLevel.compareTo(currentCredentialStrength) > 0) {
+            orchSessionService.updateSession(
+                    orchSession.withCurrentCredentialStrength(lowestRequestedCredentialTrustLevel));
```

Comment:
shared session isn't saved at this point, but later in code just before the handler returns. This orchSession save happens in a different `try` block even, so orchSession could get updated while the shared session doesn't get updated. It also means orchSession gets updated with a currentCredentialStrength even if this handler fails in the last `try` block point. Perhaps best to save at the same point as shared session saves here I reckon.

Reply:
Good point. Agree.